uplink/piecestore/upload.go

```diff
@@ -138,7 +145,14 @@ func (client *Upload) Write(data []byte) (written int, err error) {
 			},
 		})
 		if err != nil {
-			err = ErrProtocol.Wrap(err)
+			cancelErr := client.Cancel(ctx)
```

Comment:
The caller must call either `Cancel` or `Commit` anyways, so you would get wrong error message afterwards from the last cancel.

Reply:
Then I misunderstood what I had to do per comment https://github.com/storj/storj/pull/2658#discussion_r308666224

Reply:
I'm still not knowing then what to change there.

Reply:
I would leave it as is.

Reply:
oh my god! that comment didn't sound like that!

I'll roll it back